src/components/MessageList.js

```diff
@@ -34,19 +36,17 @@ class MessageList extends PureComponent {
     this.notificationTimeouts = [];
   }
   static propTypes = {
-    /** The attachment component to render, defaults to Attachment */
-    Attachment: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
-    /** A list of immutable messages */
-    messages: PropTypes.array.isRequired,
-    /** Via Context: The channel object */
-    channel: PropTypes.object.isRequired,
-    /** Via Context: The function to update a message, handled by the Channel component */
-    updateMessage: PropTypes.func.isRequired,
-    /** Via Context: The function is called when the list scrolls */
-    listenToScroll: PropTypes.func,
-    /** Typing indicator component to render  */
+    /**
+     * Typing indicator UI component to render
+     *
+     * Defaults to and accepts same props as: [TypingIndicator](https://github.com/GetStream/stream-chat-react/blob/master/src/components/TypingIndicator.js)
+     * */
     TypingIndicator: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
-    /** Date separator component to render  */
+    /**
+     * Date separator UI component to render
+     *
+     * Defaults to and accepts same props as: [DateSeparator](https://github.com/GetStream/stream-chat-react/blob/master/src/components/DateSeparator.js)
+     * */
     dateSeparator: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
```

Comment:
This should probably be DateSeparator as it's a UI component. But that might just be a breaking change. :/

Reply:
maybe removing the this.props.Message.someMethod could be part of that as well.